plutarch-test/src/Plutarch/BoolSpec.hs

```diff
@@ -0,0 +1,64 @@
+module Plutarch.BoolSpec (spec) where
+
+import Test.Syd
+
+import Plutarch
+import Plutarch.Bool
+import Plutarch.Test
+
+spec :: Spec
+spec = do
+  describe "bool" $ do
+    describe "pnot" $ do
+      goldens
+        All
+        [ ("lam", popaque pnot)
+        , ("app", popaque $ pnot #$ pcon PTrue)
+        ]
+      it "true" $ (pnot #$ pcon PTrue) #@?= pcon PFalse
+      it "false" $ (pnot #$ pcon PFalse) #@?= pcon PTrue
+    describe "pand" $ do
+      goldens
+        All
+        [ ("tf", popaque $ pcon PTrue #&& pcon PFalse)
+        , ("ft", popaque $ pcon PFalse #&& pcon PTrue)
+        , ("tt", popaque $ pcon PTrue #&& pcon PTrue)
+        , ("ff", popaque $ pcon PFalse #&& pcon PFalse)
+        ]
+      it "tf" $ (pcon PTrue #&& pcon PFalse) #@?= pcon PFalse
+      it "ft" $ (pcon PFalse #&& pcon PTrue) #@?= pcon PFalse
+      it "tt" $ (pcon PTrue #&& pcon PTrue) #@?= pcon PTrue
+      it "ff" $ (pcon PFalse #&& pcon PFalse) #@?= pcon PFalse
+      describe "laziness" $ do
+        let p1 = pand # pcon PFalse # pdelay perror
+            p2 = pcon PFalse #&& perror
+        goldens All [("pand", popaque p1), ("op", popaque p2)]
+        it "pand" $ passert $ pnot # pforce p1
+        it "op" $ passert $ pnot # p2
+        it "pand.perror" $ do
+          pfails $ pand # pcon PFalse # perror
+          pfails $ pand # pcon PTrue # perror
+          pfails $ pcon PTrue #&& perror
+    describe "por" $ do
```

Comment:
While not a blocking issue, I do wonder how we could make the testing API more ergonomic.

Reply:
I'll think about this outside this PR. Maybe some sort of MonadWriter that collects programs on the goal, and batches golden tests at the end ...